optimism/src/keccak/environment.rs

```diff
@@ -30,7 +30,7 @@ pub struct KeccakEnv<Fp> {
     /// How many blocks are left to absrob (including current absorb)
     pub(crate) blocks_left_to_absorb: u64,
     /// What step of the hash is being executed (or None, if just ended)
-    pub(crate) curr_step: Option<KeccakStep>,
+    pub(crate) curr_step: (u64, Option<KeccakStep>),
```

Comment:
Nit: would be nice to separate these with names, I don't think they benefit from coupling